src/main/java/uk/gov/justice/digital/delius/service/ConvictionService.java

```diff
@@ -338,6 +342,55 @@ public Custody addOrReplaceOrDeleteCustodyKeyDates(Long offenderId, Long convict
                 .getCustody());
     }
 
+    public Optional<ProbationStatusDetail> probationStatusFor(String crn) {
+        return offenderRepository.findByCrn(crn)
+            .filter(offender -> offender.getSoftDeleted() == 0L)
+            .map((offender) -> new ProbationStatusDetail(
+                probationStatusOf(offender),
+                previouslyKnownTerminationDateOf(offender),
+                inBreachOf(offender),
+                preSentenceActivityOf(offender)));
+    }
+
+    private ProbationStatus probationStatusOf(Offender offender) {
+        if (offender.getCurrentDisposal() == 1)
+            return ProbationStatus.CURRENT;
+        if (offender.getEvents().stream().anyMatch(event -> event.getDisposal() != null))
+            return ProbationStatus.PREVIOUSLY_KNOWN;
+        return ProbationStatus.NOT_SENTENCED;
+    }
+
+    private LocalDate previouslyKnownTerminationDateOf(Offender offender) {
+        if (!probationStatusOf(offender).equals(ProbationStatus.PREVIOUSLY_KNOWN))
+            return null;
+        return Optional.ofNullable(offender.getEvents())
+            .orElse(Collections.emptyList())
+            .stream()
+            .filter(conviction -> conviction.getDisposal() != null && conviction.getDisposal().getTerminationDate() != null)
+            .map(conviction -> conviction.getDisposal().getTerminationDate())
+            .max(Comparator.comparing(LocalDate::toEpochDay))
+            .orElse(null);
+    }
+
+    private Boolean inBreachOf(Offender offender) {
+        if (!probationStatusOf(offender).equals(ProbationStatus.CURRENT))
+            return null;
+        return Optional.of(offender)
+            .map(o ->
+                activeEvents(o.getEvents())
+                    .stream()
+                    .anyMatch(event -> event.getInBreach() == 1L)
```

Comment:
Wondering if it might be better to use TypesTransformer.convertToBoolean here to provide more readable / natural version of this ?